test/catch2/misc_tests/test_session_get_last_error.cpp

```diff
@@ -0,0 +1,46 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include <catch2/catch.hpp>
+#include "wt_internal.h"
+#include "../wrappers/mock_session.h"
+
+/*
+ * [session_get_last_error]: test_session_get_last_error.cpp
+ * Tests the API for getting verbose information about the last error of the session
+ */
+
+static const char *home;
+
+TEST_CASE("Session get last error - test getting verbose info about the last error in the session",
+  "[session_get_last_error]")
+{
+    WT_CONNECTION *conn;
+    WT_SESSION *session;
+
+    /* Open a connection to the database, creating it if necessary. */
+    REQUIRE(wiredtiger_open(home, NULL, "create", &conn) == 0);
+    REQUIRE(conn->open_session(conn, NULL, NULL, &session) == 0);
+
+    SECTION("Test API placeholder")
+    {
+        WT_SESSION_IMPL *session_impl = (WT_SESSION_IMPL *)session;
+
+        /* Prepare return arguments */
+        int err, sub_level_err;
+        char *err_msg;
```

Comment:
If we change
char *err_msg;
to
const char *err_msg;

Then the cast in
`__ut_session_get_last_error(session_impl, &err, &sub_level_err, (const char **)&err_msg);`
is no longer required.

Reply:
Actually I just requested a change to `char*` (I didn't see this comment earlier), since the code allocates memory for the buffer via `__wt_malloc` and uses `strcpy`, so `const char*` didn't seem appropriate in that context.

Jeremy, why do you think this should be `const char*`?

Reply:
+1, I'd also like some clarification on what I should be using here

Reply:
I've changed it to plain `char *` in [this commit](https://github.com/wiredtiger/wiredtiger/pull/11327/commits/b4274a2db04d7a29a134df9f6693fa23333fbd02).